clang/lib/Sema/SemaFunctionEffects.cpp

```diff
@@ -1028,9 +1026,11 @@ class Analyzer {
             return;
           }
         }
-        // If the callee is both noreturn and noexcept, it presumably
+        // If the callee is both `noreturn` and `noexcept`, it presumably
         // terminates. Ignore it for the purposes of effect analysis.
-        if (FD->isNoReturn() && isNoexcept(FD))
+        // If not C++, `noreturn` alone is sufficient.
```

Comment:
Maybe expand this comment to include that there are some C functions that are `noreturn`, but block, but that most of the relevant ones are builtins and thus already handled above.